drivers/Serial.h

```diff
@@ -53,15 +54,20 @@ namespace mbed {
  * }
  * @endcode
  */
-class Serial : public SerialBase, public Stream, private NonCopyable<Serial> {
+class
+    MBED_DEPRECATED_SINCE(
+        "mbed-os-6.0.0",
+        "Use UnbufferedSerial instead."
```

Comment:
Most people probably shouldn't be using `UnbufferedSerial`. Any `RawSerial` users probably want to move to it, but I'd bet 90% of `Serial` users are actually just using it to access the console, as that's all it's useful for.

* If being used just to access the console, use C library functions such as `printf` and `puts` instead. (Then configuration like baud rate, flow control or buffering can be manipulated via JSON).
* If being used for normal blocking application use to something other than the console, `BufferedSerial` will perform better, causing less CPU load and latency issues.
* Only if they're short of RAM and really can't afford buffering, or they've done something to bypass the locks to use this from IRQ should they shift to `UnbufferedSerial`.

Not sure how to summarise that in a little "deprecated" thing though.

Reply:
`"Use the re-targeted printf and puts instead to access the console, BufferedSerial for blocking applications or UnbufferedSerial if bypassing locks in IRQ or short of RAM"`
How about that?

Reply:
Something like that. I think the word "re-targeted" isn't going to mean much to normies, so leave it out.

We should also point to some docs covering the issue, when/if they exist. There's the recent serial design doc, for example?

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/9dcd787328d3dbbf521db5a4de3ec4bc055d5895..1dc9c09088474e7f76830bd12befb2620845e782).